docx_editor/track_changes.py

```
@@ -169,50 +179,129 @@ def batch_edit(self, operations: list[EditOperation]) -> list[int]:
+        """Validate op's action-specific required args and return the search target.
+
+        Shared by ``_apply_single_edit`` and ``_validate_single`` so the two
+        paths cannot drift out of sync.
 
+        Raises:
+            ValueError: If required arguments for op.action are missing, or the
+                action is unrecognized.
+        """
         if op.action == "replace":
             if not op.find or op.replace_with is None:
                 raise ValueError("replace requires 'find' and 'replace_with'")
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)